### PR TITLE
[Test Framework] Remove dependency on kubectl library

### PR DESCRIPTION
**Pull request by nmittler**
The Accessor apply and delete commands currently rely on the kubectl utilities for these functions. This change removes this unnecessary dependency and uses the underlying kube cliruntime libraries directly.

I initially expected to be able to reuse code from istioctl, but it appears that they rely on a helm-style deployment. It may be possible to use some part of helm to do the raw k8s deploy, but nothing stood out. If the code I have here works, it's pretty straight-forward anyway.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by nmittler** [comment]
FYI for now, I'm just testing the apply logic. Once this is working, I'll follow up with delete.

File: pkg/test/kube/accessor.go [hunk 1]
```diff
@@ -980,3 +988,67 @@ func (d *yamlDoc) toTempFile(workDir, fileNamePrefix string) (string, error) {
...
+func newDiscoveryClient(restConfig *rest.Config) (discovery.CachedDiscoveryInterface, error) {
+	// Copy the rest config.
+	cfg := *restConfig
+
+	// The more groups you have, the more discovery requests you need to make.
+	// given 25 groups (our groups + a few custom resources) with one-ish version each, discovery needs to make 50 requests
+	// double it just so we don't end up here again for a while.  This config is only used for discovery.
+	cfg.Burst = 100
+
+	parentDir := filepath.Join(homedir.HomeDir(), ".kube", "cache", "discovery")
```
Comment (howardjohn) [inline pkg/test/kube/accessor.go:1031]: this is a bit suspicious, is there not a higher level part of the k8s library we can rely on?

**Comment by nmittler** [comment]
@howardjohn this should be ready to go now. Mind taking a look?

File: pkg/test/kube/apply.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+	// Get the existing object from the server.
+	if err := info.Get(); err != nil {
+		if !kubeErrors.IsNotFound(err) {
+			return addSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%s\nfrom server for:", info.String()), info.Source, err)
+		}
+
+		// Create the resource if it doesn't exist
+		// First, update the annotation used by kubectl apply
+		if err := kube.CreateApplyAnnotation(info.Object, unstructured.UnstructuredJSONScheme); err != nil {
```
Comment (howardjohn) [inline pkg/test/kube/apply.go:186]: do we actually need this?

File: pkg/test/kube/apply.go [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+			// Concatenate the stdout and stderr
+			s := stdout.String() + stderr.String()
+			scopes.Framework.Infof("Failed applying YAML file: %s (err: %v): %s", file, err, s)
+			return fmt.Errorf("%v: %s", err, s)
+		}
+	}
+	return nil
+}
+
+func (a *accessorImpl) applyObject(info *resource.Info, s *openAPISchema, dryRun bool, stdout io.Writer,
```
Comment (howardjohn) [inline pkg/test/kube/apply.go:159]: This is not really maintainable... If we are going to try to literally be kubectl we should probably use the kubectl library. If we aren't then we can make things much simpler I think. I would expect we can just do 
```
cl.Patch(context.TODO(), us.GetName(), types.ApplyPatchType, bytes, metav1.PatchOptions{
                                FieldManager: "istio-testing",
                                Force:        util.BoolPointer(true),
                        })
```

and bypass most of this code

File: pkg/test/kube/delete.go [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+}
+
+func (a *accessorImpl) deleteInternal(namespace string, files []string) (err error) {
+	for _, f := range removeEmptyFiles(files) {
+		err = multierror.Append(err, a.deleteFile(namespace, f)).ErrorOrNil()
+	}
+	return err
+}
+
+func (a *accessorImpl) deleteFile(namespace string, file string) error {
```
Comment (howardjohn) [inline pkg/test/kube/delete.go:91]: same comment as above, why not just `dynamicClient.Delete(name, namespace)`?

**Review by howardjohn — COMMENTED** [review]
I am not sure I understand the motivation here. It looks like we have just copy+pasted a large chunk of kubectl. What do we gain? We still indirectly depend on kubectl btw, so our dependencies are not reduced

File: operator/pkg/helmreconciler/wait.go [hunk 5]
```diff
@@ -12,11 +12,28 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/*
+Copyright 2016 The Kubernetes Authors.
```
Comment (hzxuzhonghu) [inline operator/pkg/helmreconciler/wait.go:16]: why add k8s copyright

**Comment by nmittler** [comment]
Closing this for now ... I'm not sure it makes sense, given that we'll still have a transitive dependency on kubectl (via helm).

**Pull request closed without merging**